### ISSUE TITLE
Spec for switch on symbols shouldn't focus on the syntax.

### ISSUE THREAD

**Issue by lrhn** [dart-lang/sdk#24469]
The specification says 

> It is a compile-time error if the class C has an implementation of the operator
> == other than the one inherited from Object unless the value of the
> expression is a string, an integer, literal symbol or the result of invoking a
> constant constructor of class Symbol."

It's not clear which "expression" is being referenced here (but it's likely each "case" expression, e_k).

This distinguishes between `#foo` and `const Symbol("foo")` - which are the same value - and makes it sound like the only _expression_ that can be written in a switch case is the literal or const invocation.

Constant values are distinguishable by value, not syntax, and they can be referenced by name, so:

```
const z = #z;
switch (x) { 
   case z:
   case const Symbol("v"):   
   case #w:
}
```

should  all be valid.

The spec only need to say:  

> ... is a string, an integer, or an instance of the class `Symbol` from the platform libraries.

It already needs to be constant, so focusing on the syntax is just confusing.

**Comment by eernstg** [comment]
No milestone now: This clarification should be performed, but it does not block Dart 2.

**Comment by eernstg** [comment]
Closing: Fixed in a12805b90ff14e0ae47a74cc3659b02b2d18841c.
